### PR TITLE
no py27 in travis

### PR DESCRIPTION
**Pull request by drowoseque**
## Description
- py27 tests removed from `.travis.yml`

## Motivation and Context
#2830 

## Have you tested this? If so, how?
Tests are passing since several envs were removed

### CODE REVIEW

**Comment by drowoseque** [comment]
@honnix @dlstadther 
can't figure out why coverage become so low on `luigi.contrib.s3`  after removing py27 env in travis
could you please have a quick look through?
have any ideas?

**Comment by honnix** [comment]
> can't figure out why coverage become so low on `luigi.contrib.s3`

What makes you believe this is low? https://codecov.io/gh/spotify/luigi/compare/34c80af2d96b7cfc1a96727a575ed6057352b416...d7acc5e120dfbafc9a86b309c4e3b689fbed74cb/diff looks fine to me.

**Comment by drowoseque** [comment]
actually there are unexpected coverage changes in the report

https://codecov.io/gh/spotify/luigi/compare/34c80af2d96b7cfc1a96727a575ed6057352b416...d7acc5e120dfbafc9a86b309c4e3b689fbed74cb/changes

File: luigi/format.py [hunk 1]
```diff
@@ -510,13 +477,7 @@ def pipe_writer(self, output_pipe):
 SysNewLine = NewlineFormat()
 Gzip = GzipFormat()
 Bzip2 = Bzip2Format()
-MixedUnicodeBytes = MixedUnicodeBytesFormat()
```
Comment (honnix) [inline luigi/format.py]: What is the replacement of this?

Comment (drowoseque) [reply luigi/format.py]: @honnix 
no replacement as this format was introduced in #786 with comment
> Accept to write unicode in Files for compat  Only accept on python2 Fix #786

Comment (honnix) [reply luigi/format.py]: Yeah it was added fo py2 compatibility, but could be abused for other purpose :( If for whatever reason, `lines` is a mixture of `str` and `bytes` and people decide to use this.

Comment (drowoseque) [reply luigi/format.py]: I don't think people use `luigi.format.*` in their mundane lives outside `luigi`
do you think we should leave it here?

Comment (honnix) [reply luigi/format.py]: I'm very sure people do and that is what these formatters for. I would suggest we leave it there and revamp the test case to reflect the "correct" usage: dealing with mixed of bytes and str. Wdyt?

Comment (drowoseque) [reply luigi/format.py]: sounds good.
+ maybe we should add deprecation warning for that

Comment (honnix) [reply luigi/format.py]: We don't need to deprecate it I think. It serves a use case and we have no replacement that we can recommend user to migrate to.

**Pull request merged**